sessions/2019/march/Session_28.md

```diff
@@ -0,0 +1,87 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 28 (19/03/2019)
+
+- Reviewing sprint project (issue) tasks and solving problems & feedback
+- Reviewing main project tasks  & planning next Sprint
+
+## Notes
+
+#### Reviewing sprint project (issue) tasks and solving problems & feedback
+
+Public classes goes into separate files, to separate it quickly we can copy the class code and paste into the package directory and IntellJ Idea creates a class for us)
+
+Always that be possible we have to make attributes `final`. This keyword, means our object, field or property is in a fixed memory position and can be seen as a constant.
```

Comment:
That's partially correct, and the benefits (thread-safety and potentially allowing compiler optimisations) are the same, but may lead to confusion as you write more code and things aren't as constant as they might appear to be.

I prefer to think of it as "It will be assigned (using `=`) exactly once". For example, you can have a `final List<String> strings = new ArrayList<>();` and still add and remove elements from the list. In fact, because of how `ArrayList` works, if you add a lot of elements or remove a lot of elements, the array list will have to allocate memory to store the additional data anyway.